x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/index.tsx

```diff
@@ -477,77 +477,90 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
     isEqlSequenceQuery(queryBar?.query?.query as string) &&
     groupByFields.length === 0;
 
-  const isSuppressionGroupByDisabled =
+  /**
+   * If we don't have ML field information, users can't meaningfully interact with these fields */
+  const areSuppressionFieldsDisabledByMlFields =
+    isMlRule(ruleType) && (noMlJobsStarted || mlFieldsLoading || !mlSuppressionFields.length);
```

Comment:
<img width="952" alt="Screenshot 2024-07-17 at 12 05 43" src="https://github.com/user-attachments/assets/7c2dfc98-8325-4a92-8448-ce012ea04a1e">

I added two jobs (one running, one not). Selected field, configured suppression and then removed running job.
So, suppression config is disabled, but field actually was from a running job.

If I save rule, with "disabled" suppression and start job - suppression would work, but the field stored there would be from another job.

 I can't clear this suppression on form , because when I removed job, it is still disabled
<img width="954" alt="Screenshot 2024-07-17 at 12 13 15" src="https://github.com/user-attachments/assets/efa91bb5-3294-4405-9e41-9b99f19100fe">

So, I have to add the same running job to clear it.

Similar can happen when user changes index, without adjusting suppression fields. But in this case, suppression controls are not disabled and user can easily change those fields
Here, after removing running job, controls become disabled. It requires user to add this job again to change fields or start another.
I am not sure, how critical it can be - just an observation on behaviour.
What do you think about it?

Reply:
Thanks for calling this out. I agree that the UX wasn't great, so I fixed it in 8b6c460e3815258a6c8c314ae5822e4d4e4ad485.